tiup/tiup-component-cluster-patch.md

```diff
@@ -0,0 +1,84 @@
+
+- Uploads the binary package for replacement to the target machine.
+- If the target service is a storage service such as TiKV, TiFlash, or TiDB Binlog, TiUP first takes the related nodes offline via the API.
+- Stops the target service.
+- Unpacks the binary package and replace the service.
+- Starts the target service.
+
+## Syntax
+
+```shell
+tiup cluster patch <cluster-name> <package-path> [flags]
+```
+
+- `<cluster-name>`: The name of the cluster to be operated.
+- `<package-path>`: The path to the binary package used for replacement. The steps to generate the package are as follows:
+    - Determine the name `${component}` of the component to be replaced (tidb, tikv, pd...), the `${version}` of the component (v4.0.0, v4.0.1 ...), and the operating system `${os}` and platform `${arch}` on which the component runs.
+    - Download the current component package using the command `wget https://tiup-mirrors.pingcap.com/${component}-${version}-${os}-${arch}.tar.gz -O /tmp/${component}-${version}-${os}-${arch}.tar.gz`.
+    - Run `mkdir -p /tmp/package && cd /tmp/package` to create a temporary directory to pack files.
+    - Run `tar xf /tmp/${component}-${version}-${os}-${arch}.tar.gz` to unpack the original binary package.
+    - Run `find .` to view the file structure in the temporary package directory.
+    - Copy the binary files or configuration files to the corresponding locations in the temporary directory.
+    - Run `tar czf /tmp/${component}-hotfix-${os}-${arch}.tar.gz *` to pack the files in the temporary directory.
+    - Finally, you can use `/tmp/${component}-hotfix-${os}-${arch}.tar.gz` as the `<package-path>` in the `tiup cluster patch` command.
+
+## Options
+
+### --overwrite
+
+- After you patch a certain component (such as TiDB or TiKV), when the tiup cluster scales out the component, TiUP uses the original component version by default. To use the version that you patch when the cluster scales out in the future, you need to specified the option `--overwrite` in the command.
+- Data type: `BOOLEAN`
+- This option is disabled by default with the `false` value. To enable this option, add this option to the command, and either pass the `true` value or do not pass any value.
+
+### --transfer-timeout
+
+- When restarting the PD or TiKV service, TiUP first migrates the leader of the node to be restarted to another node. Because the migration process takes some time, you can use the option `--transfer-timeout` to set the maximum waiting time (in seconds). After the timeout, TiUP directly restarts the service.
+- Data type: `UINT`
+- If this option is not specified, TiUP directly restarts the service after waiting for `300` seconds.
+
+> **Note:**
+>
+> If TiUP directly restarts the service after the timeout, the service performance may jitter.
```

Comment:
I think "跳过等待" here does not actually mean skipping the waiting time. Even if this option is not specified, TiUP still waits for 300 seconds by default (see L44). 

Here's my understanding of the note:
If the migration of the leader node has not finished within the waiting time, TiUP directly restarts the service, and because TiUP directly restarts the service before the migration of the leader node finishes, the service performance might jitter.

Reply:
在滚动重启 TiKV 的过程中，会有 transfer leader 的阶段，如果 transfer leader 没有结束就强制重启了 TiKV，可能会性能抖动，但是 TiKV 并不保证在多久之内一定 transfer 完所有 leader，所以这里的 timeout 是说 tiup 等多久 tikv transfer leader，如果这个时间过去了还没搞定，就直接重启不等了。所以 @CharLotteiu 你的理解没问题。

Reply:
```suggestion
> If TiUP directly restarts the service after the timeout, the service performance might jitter.
```